data/src/emulator.js

```diff
@@ -5671,34 +5882,57 @@ class EmulatorJS {
     }
 
     screenRecord() {
-        const captureScreenWidth = (this.config.screenRecording && (typeof this.config.screenRecording.width == "number")) ? this.config.screenRecording.width : 800;
-        const captureScreenHeight = (this.config.screenRecording && (typeof this.config.screenRecording.height == "number")) ? this.config.screenRecording.height : 600;
-        const captureFps = (this.config.screenRecording && (typeof this.config.screenRecording.fps == "number")) ? this.config.screenRecording.fps : 30;
-        const captureVideoBitrate = (this.config.screenRecording && (typeof this.config.screenRecording.videoBitrate == "number")) ? this.config.screenRecording.videoBitrate : 2 * 1024 * 1014;
-        const captureAudioBitrate = (this.config.screenRecording && (typeof this.config.screenRecording.audioBitrate == "number")) ? this.config.screenRecording.audioBitrate : 256 * 1024;
+        const captureFps = this.preGetSetting("screenRecordingFPS") || this.config.screenRecording.fps;
+        const captureFormat = this.preGetSetting("screenRecordFormat") || this.config.screenRecording.format;
+        const captureQuality = this.preGetSetting("screenRecordQuality") || this.config.screenRecording.quality;
+        const captureVideoBitrate = this.preGetSetting("screenRecordVideoBitrate") || this.config.screenRecording.videoBitrate;
+        const captureAudioBitrate = this.preGetSetting("screenRecordAudioBitrate") || this.config.screenRecording.audioBitrate;
+        const aspectRatio = this.gameManager.getVideoDimensions("aspect") || 1.333333;
+        const videoRotation = parseInt(this.preGetSetting("videoRotation") || 0);
```

Comment:
Why are we using `preGetSetting`? This should only be used before the game starts (since this always pulls directly from localstorage). Pull it from `this.settings` instead. This should also probably be combined into that `this.capture.video` object